### ISSUE TITLE
Spirit Shroud base 1d8 damage (when cast at lvl 3) not being applied

### ISSUE THREAD

**Issue by kosocku** [SolastaMods/SolastaUnfinishedBusiness#2134]
Game version: 1.4.33 Public
Mod version 1.4.33.0

Bug: When casting Spirit Shroud at lvl 3, the spell should be adding 3 things: 1) a Speed reduction of 10 ft to anybody starting it's turn within the 10-ft radius of the spell until the start of it's next turn, 2) a debuff that prevents a target that's hit while the spell is active from regaining hit points until the start of your next turn, and 3) 1d8 of damage which can be radiant, necrotic, or cold (chosen when you cast the spell) to any target you hit that's within the 10-ft radius of the spell. The 1d8 damage is not being applied for any of the 3 damage types (I tried casting twice, choosing first radiant and then cold on the 2nd cast), even though other spell effects (slowing and prevention from healing) seem to be working fine. (Note: Was working in the AM of 2023-01-05 before the update to 1.4.33.0, but not after the update)

Recreating should be easy, as I've tried to cast the spell - both in and before starting combat - and the same thing happens.

Attached is a chat log of the battle, showing there was NO cold damage (which I'd chosen when I cast the spell) being done on a hit, even thought the spell was active.
![IMG_0766](https://user-images.githubusercontent.com/122100625/211064600-95b2ed45-e45a-4fe1-8d71-c10e7a5d15f1.jpg)
[UB Bug reports.zip](https://github.com/SolastaMods/SolastaUnfinishedBusiness/files/10361889/UB.Bug.reports.zip)

**Comment by ThyWoof** [comment]
![image](https://user-images.githubusercontent.com/13087646/211105235-4abf6aa2-bf5f-4024-8484-f55c71dca5e0.png)
this is a very strange issue. Did further tests and it only happens if the repertoire spell owner is a Warlock repertoire. Above I have spell on both Warlock and Sorcerer repertoires. Just casted from Sorcerer using a Warlock slot and works.

for some unknown reason when cast from Warlock I'm getting a slot level 1 instead of 3. It might be a game bug as this wasn't happening before.

**Comment by ThyWoof** [comment]
```
  private void OnActivateStandardBox(int index)
  {
    if (this.spellCastEngaged == null)
      return;
    SpellDefinition spellDefinition = this.spellsByIndex[index];
    int slotLevel = spellDefinition.SpellLevel;
    if (!this.spellRepertoire.IsMysticArcanumSpell(spellDefinition) && this.spellRepertoire.SpellCastingFeature.UniqueLevelSlots && slotLevel > 0)
      slotLevel = this.spellRepertoire.GetLowestAvailableSlotLevel();
    if (spellDefinition.SpellsBundle)
    {
      SubspellSelectionModal screen = Gui.GuiService.GetScreen<SubspellSelectionModal>();
      screen.Bind(spellDefinition, this.caster, this.spellRepertoire, this.spellCastEngaged, slotLevel, this.activationBoxes[index].RectTransform);
      screen.Show();
    }
    else
      this.spellCastEngaged(this.spellRepertoire, spellDefinition, slotLevel);
  }
```

this only happens with MC Warlocks as far as I can tell. Will need to think wisely the best way to bypass this as well as other bugs I've found that use same pattern.

**Comment by ThyWoof** [comment]
I believe I got all bases covered including Invocations that consume slots.

![image](https://user-images.githubusercontent.com/13087646/211122423-3905dd63-01be-4318-b365-1dd60aa1310f.png)
Upcasting from Warlock repertoire works [correctly used level V Sorcerer slot and did 2 dice damage]

![image](https://user-images.githubusercontent.com/13087646/211122548-305c5337-6e65-4457-bd06-a8827cdd851d.png)
No-Upcasting from Warlock repertoire now works as well.

![image](https://user-images.githubusercontent.com/13087646/211122584-56523492-b519-4f26-8367-196aaef58be0.png)
Correctly consumed the Warlock slot on this case.

Fixed on [DEV](https://github.com/SolastaMods/SolastaUnfinishedBusiness/commit/3601ab486d0b7cfd2e4768116f2b69d76577c227).
